conda_libmamba_solver/state.py

```diff
@@ -402,10 +403,14 @@ def channels_from_installed(self) -> Iterable[Channel]:
                 # These "channels" are not really channels, more like
                 # metadata placeholders
                 continue
-            subdir_url = record.channel.subdir_url
-            if subdir_url not in seen_urls:
-                seen_urls.add(subdir_url)
-                yield record.channel
+            if record.channel.base_url is None:
+                continue
+            if record.channel.subdir_url in seen_urls:
+                continue
+            if is_channel_available(record.channel.base_url):
```

Comment:
we intend to skip any channels that are unavailable right?

```suggestion
            if not is_channel_available(record.channel.base_url):
```